src/lex.cpp

```diff
@@ -1288,6 +1292,52 @@ void lexer::parse_modern_octal_number() {
       diag_unexpected_characters_in_octal_number>(input);
 }
 
+void lexer::check_precision_loss(const char8* number_begin,
+                                 const char8* input) {
+  string8_view number_literal(number_begin,
+                              static_cast<size_t>(input - number_begin));
+  const size_t GUARANTEED_ACC_LENGTH = 15;
+  const size_t MAX_ACC_LENGTH = 309;
+  if (number_literal.size() > GUARANTEED_ACC_LENGTH) {
```

Comment:
Nit: Instead of wrapping the whole function's body in an `if`, use an early return:
```suggestion
  if (number_literal.size() <= GUARANTEED_ACC_LENGTH) {
    return;
  }
```